Makefile

```diff
@@ -0,0 +1,313 @@
+SHELL:=/bin/bash
```

Comment:
I would put all of this in for example `dev/` directory, and it would be good that we have README, with explanation how to setup environment and dependencies.

Reply:
if its in a `dev/` dir, it will break many of the file paths and in general make it awkward to use the Makefile and a lot more complicated to configure it. 

The `export` lines show all the environment variables, and the `install` recipe shows the dependency installation, along with the details in the `help` recipe

Reply:
Maybe moving it to the different directory is not the best idea, but I wanted to achieve something like we did with singularity [here](https://github.com/mskcc/beagle/tree/develop/container). Also by README I was thinking on readme similar to one in previous link. Just something which can help user to run beagle using this method as soon as possible without having to read the whole Makefile.

Reply:
I am updating the [`help`](https://github.com/mskcc/beagle/blob/5f793d976fa56ad10c1e866d5ae4cd1470839cea/Makefile#L7) message to make this more clear. `make help` lists the commands needed to set up and run Beagle. Since the Makefile will include all the required environment variables (or as many as possible; `help` will list non-included extras) it will not be required for the user to read the whole thing to get it working. Things like [`check-port-collision`](https://github.com/mskcc/beagle/blob/5f793d976fa56ad10c1e866d5ae4cd1470839cea/Makefile#L282) are also included for the user to quickly troubleshoot issues with services.